docs_app/content/guide/core-semantics.md

```diff
@@ -0,0 +1,25 @@
+# RxJS Core Semantics
+
+Starting in version 8, all RxJS operators that are provided in the core library MUST meet the following semantics. In the current version, version 7, all operators SHOULD meet the following semantics (as guidelines). If they do not, we need to track the issue on [GitHub](https://github.com/ReactiveX/rxjs/issues).
+
+## Purpose
+
+The purpose of these semantics is provide predictable behavior for the users of our library, and to ensure consistent behavior between our many different operators.
+
+## Operators
+
+    - MUST be a function that returns an [operator function](https://rxjs.dev/api/index/interface/OperatorFunction). That is `(source: Observable<In>) => Observable<Out>`.
+    - The observable returned by the operator function MUST subscribe to the source.
+    - If the operation performed by the operator can tell it not change anything about the output of the source, it MUST return the reference to the source. For example `take(Infinity)` or `skip(0)`.
```

Comment:
nit: ...can tell it **does** not change... I had trouble parsing this sentence without that